### PR TITLE
[logging] removed almost all  System.out references and replaces with…

### PR DESCRIPTION
**Pull request by cboddy**
… java.util.logging

Moving  to a  logging framework from raw System.out logging has several
 benefits which  will make easier to understand what is going on when reading
 the logs ie.
 * instrumented log messages
 * timestamped log messages

 This was mainly done by regex, for details see
 https://gist.github.com/cboddy/31874bcdc68d06a514eefa2f2e1b4cec

### CODE REVIEW

**Comment by ianopolous** [comment]
Have you tested this out in the browser to see the logging it the same?

**Comment by cboddy** [comment]
@ianopolous yep, attached is the browser web-console logging in with and without this change.
[console_with_sout.log](https://github.com/Peergos/Peergos/files/2152640/console_with_sout.log)
[console_with_logger.log](https://github.com/Peergos/Peergos/files/2152641/console_with_logger.log)

**Comment by ianopolous** [comment]
Can you resolve the conflict?

File: src/peergos/server/Command.java [hunk 1]
```diff
@@ -60,7 +62,7 @@ public void main(Args args) {
         if (headOpt.isPresent()) {
             String head = headOpt.get();
             if (head.equals("help")) {
-                System.out.println(helpMessage());
+                LOG.info(helpMessage());
```
Comment (ianopolous) [inline src/peergos/server/Command.java]: I think the idea with the cmd interface is you print exactly what you want, so we wouldn't want the logger here.

Comment (ianopolous) [reply src/peergos/server/Command.java]: same with all the other commands

**Comment by cboddy** [comment]
And rebased to resolve  the merge conflict.

File: src/peergos/server/Downloader.java [hunk 2]
```diff
@@ -13,6 +14,7 @@
 import java.util.function.*;
 
 public class Downloader {
+	private static final Logger LOG = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
```
Comment (ianopolous) [inline src/peergos/server/Downloader.java]: why not Logger.getGlobal?

Comment (cboddy) [reply src/peergos/server/Downloader.java]: Sure, feel free to update it

Comment (ianopolous) [reply src/peergos/server/Downloader.java]: I don't have time for that. I barely have time to manually review every change here.

File: src/peergos/server/Downloader.java [hunk 3]
```diff
@@ -27,7 +29,7 @@ public static void main(String[] args) throws Exception {
         long t1 = System.currentTimeMillis();
         downloadTo(context, fromPath, Paths.get(toPath), props -> true, pool);
         long t2 = System.currentTimeMillis();
-        System.out.println("Download took " + (t2-t1) + " mS");
+        LOG.info("Download took " + (t2-t1) + " mS");
```
Comment (ianopolous) [inline src/peergos/server/Downloader.java]: This is designed as a cmd line class, so this should be System.out

File: src/peergos/server/Login.java [hunk 4]
```diff
@@ -15,6 +17,6 @@ public static void main(String[] args) throws Exception {
         Console console = System.console();
         String password = new String(console.readPassword("Enter password for " + username + ":"));
         UserContext context = UserContext.signIn(username, password, network, crypto).get();
-        System.out.println("Logged in " + username + " successfully!");
+        LOG.info("Logged in " + username + " successfully!");
```
Comment (ianopolous) [inline src/peergos/server/Login.java]: This is designed as a cmd line class, so this should be System.out

File: src/peergos/server/Renew.java [hunk 5]
```diff
@@ -18,6 +20,6 @@ public static void main(String[] args) throws Exception {
         String password = new String(console.readPassword("Enter password for " + username + ":"));
         UserContext context = UserContext.signIn(username, password, network, crypto).get();
         context.renewUsernameClaim(expiry).get();
-        System.out.println("Logged in " + username + " successfully!");
+        LOG.info("Logged in " + username + " successfully!");
```
Comment (ianopolous) [inline src/peergos/server/Renew.java]: This is designed as a cmd line class, so this should be System.out

File: src/peergos/server/RenewUsernameClaim.java [hunk 6]
```diff
@@ -27,6 +29,6 @@ public static void main(String[] args) throws Exception {
         UserContext context = UserContext.signIn(username, password, network, crypto).get();
         boolean isExpired = context.usernameIsExpired().get();
         if (isExpired)
-            System.out.println(context.renewUsernameClaim(expiry).get() ? "Renewed username" : "Failed to renew username");
+            LOG.info(context.renewUsernameClaim(expiry).get() ? "Renewed username" : "Failed to renew username");
```
Comment (ianopolous) [inline src/peergos/server/RenewUsernameClaim.java]: This is designed as a cmd line class, so this should be System.out

File: src/peergos/server/Start.java [hunk 7]
```diff
@@ -25,8 +25,8 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-public class Start
-{
+public class Start {
```
Comment (ianopolous) [inline src/peergos/server/Start.java]: This whole class is based around shell commands that don't want ugly logging wrapping what they print to the user.

File: src/peergos/server/Uploader.java [hunk 8]
```diff
@@ -29,7 +31,7 @@ public static void main(String[] args) throws Exception {
         long t1 = System.currentTimeMillis();
         uploadTo(context, Paths.get(fromPath), toPath, context.fragmenter(), props -> true, pool);
         long t2 = System.currentTimeMillis();
-        System.out.println("Upload took " + (t2-t1) + " mS");
+        LOG.info("Upload took " + (t2-t1) + " mS");
```
Comment (ianopolous) [inline src/peergos/server/Uploader.java]: cmd app

File: src/peergos/server/UserStats.java [hunk 9]
```diff
@@ -30,11 +32,11 @@ public static void main(String[] args) throws Exception {
                         total += network.dhtClient.getRecursiveBlockSize(target.get()).get();
                 }
                 String summary = "User: " + username + ", expiry: " + expiry + " usage: " + total + "\n";
-                System.out.println(summary);
+                LOG.info(summary);
```
Comment (ianopolous) [inline src/peergos/server/UserStats.java]: cmd app

File: src/peergos/shared/NetworkAccess.java [hunk 10]
```diff
@@ -198,7 +200,7 @@ public NetworkAccess clear() {
                     .thenCompose(blobHash -> tree.put(writer, location.getMapKey(), metadata.committedHash(), blobHash)
                             .thenApply(res -> blobHash));
         } catch (Exception e) {
-            System.out.println(e.getMessage());
+            LOG.info(e.getMessage());
```
Comment (ianopolous) [inline src/peergos/shared/NetworkAccess.java]: looks like this one could be a severe

File: src/peergos/shared/crypto/hash/ScryptJava.java [hunk 11]
```diff
@@ -8,6 +9,7 @@
 import peergos.shared.user.*;
 
 public class ScryptJava implements LoginHasher {
+	private static final Logger LOG = Logger.getGlobal();
```
Comment (ianopolous) [inline src/peergos/shared/crypto/hash/ScryptJava.java]: too much indentation?

**Pull request merged**